py3status/modules/keyboard_layout.py

```diff
@@ -63,11 +36,16 @@ def __init__(self):
         find the best implementation to get the keyboard's layout
         """
         try:
-            xkblayout()
+            self._xkblayout()
+            self.command = self._xkblayout
         except:
-            self.command = setxkbmap
-        else:
-            self.command = xkblayout
+            self.command = self._xset
+            self.layouts = self._get_layouts()
+
+        if not self.color:
```

Comment:
this whole block and the `self.colors_dict` generation should be put on the `keyboard_layout` method because the class is instance is created by py3status **before** it can override the users' config values. This means that for now the users' config is not taken into account in your module.